pact/Pact/Core/IR/Term.hs

```diff
@@ -222,6 +222,8 @@ data Step name ty builtin info
   | StepWithRollback
     (Term name ty builtin info)
     (Term name ty builtin info)
+  | LegacyStepWithEntity (Term name ty builtin info) (Term name ty builtin info)
```

Comment:
Note: These are separate cases for a few reasons:
- We don't want to change hashes this close to release, and having new ADT cases ensures our CBOR repr doesn't have to.
- It makes it more explicit to handle cases in functions like `hasRollback` and `ordinaryDefPactStepExec`